### PR TITLE
[WFCORE-379 WFCORE-380] Operations to browse and read content from the content repository

### PR DESCRIPTION
**Pull request by mjurc**
* Adding browse-content and read-content operations to workflow of DeploymentTestCase and ExplodedDeploymentTestCase.
* Adding negative tests for browse-content and read-content operations to DeploymentOperationsTestCase.

### CODE REVIEW

**Comment by bstansberry** [comment]
@ehsavoie please approve

File: testsuite/standalone/src/test/java/org/wildfly/core/test/standalone/mgmt/api/DeploymentTestCase.java [hunk 1]
```diff
@@ -589,34 +661,79 @@ public void initialDeploy() throws IOException {
...
-                    // .dodpeloy put down by initialDeploy(). So pause a bit to
-                    // let that complete
-                    // so we don't end up having our own file deleted
+                    // full replace with single property changed
                     final File dodeploy = new File(deployDir, "test-deployment.jar.dodeploy");
                     final File isdeploying = new File(deployDir, "test-deployment.jar.isdeploying");
+                    File testDeployment = new File(deployDir, "test-deployment.jar");
+                    File propertiesFile = new File(testDeployment, "service-activator-deployment.properties");
+                    FileOutputStream os = new FileOutputStream(propertiesFile.getAbsolutePath());
+                    Properties replacementProps = new Properties();
```
Comment (ehsavoie) [inline testsuite/standalone/src/test/java/org/wildfly/core/test/standalone/mgmt/api/DeploymentTestCase.java]: Maybe you should use a try-with-resources

File: testsuite/standalone/src/test/java/org/wildfly/core/test/standalone/mgmt/api/DeploymentTestCase.java [hunk 2]
```diff
@@ -589,34 +661,79 @@ public void initialDeploy() throws IOException {
...
+                        String failureDescription = Operations.getFailureDescription(response).toString();
+                        Assert.assertTrue("Operation browse-content should fail with WFLYSRV0255, but failed with " + failureDescription,
+                                failureDescription.contains("WFLYSRV0255"));
+                    } catch (InterruptedException e) {
+                        Thread.currentThread().interrupt();
+                        throw new RuntimeException(e);
+                    } catch (ExecutionException e) {
+                        throw new RuntimeException(e.getCause());
+                    } catch (TimeoutException e) {
+                        throw new RuntimeException(e);
```
Comment (ehsavoie) [inline testsuite/standalone/src/test/java/org/wildfly/core/test/standalone/mgmt/api/DeploymentTestCase.java]: cancel the future

File: testsuite/standalone/src/test/java/org/wildfly/core/test/standalone/mgmt/api/DeploymentTestCase.java [hunk 3]
```diff
@@ -801,12 +929,167 @@ private static void cleanFile(File toClean) {
...
+            Assert.assertTrue("Expected files not listed by /deployment=test-deployment.jar:browse-content(depth="
+                    + depth + ", archive=" + archive + ") : " + expectedContents.toString(),
+                    expectedContents.isEmpty());
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new RuntimeException(e);
+        } catch (ExecutionException e) {
+            throw new RuntimeException(e.getCause());
+        } catch (TimeoutException e) {
+            throw new RuntimeException(e);
```
Comment (ehsavoie) [inline testsuite/standalone/src/test/java/org/wildfly/core/test/standalone/mgmt/api/DeploymentTestCase.java]: cancel future

**Comment by mjurc** [comment]
@ehsavoie Fixes implemented

**Comment by ehsavoie** [comment]
@bstansberry i'm ok with it

**Pull request merged**